### ISSUE TITLE
Feature request: limit workload based on the generated noise level

### ISSUE THREAD

**Issue by andrasfuchs** [FoldingAtHome/fah-client-bastet#11]
I talked to some of my friends and I convinced a few to start running FAH on their (usually fairly powerful) computers at home. They really like the idea and they wanted to help, but they stopped using it in a few days, because they complained about the noise the computer started to make.

It would be great to have one of the following options to set the maximum CPU and GPU payloads in a way that effectively would limit the noise that the computer generates:
(a) the user would define the maximum temperature of CPUs and GPUs in the system
(b) the user would define the maximum RPM of the fans of the system
(c) the user would define the maximum power used by each CPU and GPU
(d) the user would define the maximum noise picked up by the microphone [I know this is just a hack, but it might work].

**Comment by jcoffland** [comment]
Listening to the microphone has security implications and would over complicate our code.  Limiting by temperature, fan RPM or workload would be more practical.

**Comment by ZexMaxwell001** [comment]
Having a Current temp on the WU slot would be nice too. it would be a way to notify the user if a PC is showing high temps due to hardware failure and the like. 
https://foldingforum.org/viewtopic.php?t=39432

**Comment by metafarion** [comment]
Yes, I definitely want this as well.  You get SOME control through selecting the number of cores in play, but I'd much rather spread the load through several cores instead of just having one at 100% as well as having finer control of system responsiveness, fan speed, etc.  This is especially true for GPU slots.

**Comment by alrcatraz** [comment]
I'm facing similar problems. I'm running FAH 8.3 on my laptop, and when I allow the client to use my GPU, my computer will shut down for overheat in not over 30min. I really recommend to add any heat or advance workload management feature into the client.

here's some information about my computer, I hope it can be helpful.
Type: ROG Strix G614JV
CPU: Intel i9-13980HX
GPU: NVDIA GeForce RTX 4060 Laptop
RAM: 16GB * 2
OS: Windows 11 Pro

**Comment by sudo-nano** [comment]
I would love to see this feature as well. This is a serious usability issue. It disqualifies anyone without a beefy cooling setup from contributing to the project, which includes every single laptop and a substantial portion of non-enthusiast desktops.

**Comment by andrasfuchs** [comment]
I haven't used FAH in a while, but there is a [pretty good fan control application](https://github.com/Rem0o/FanControl.Releases) for Windows, which supports plugins.

As far as I understand it, it would be possible with [its plugins support](https://github.com/Rem0o/FanControl.Releases/wiki/Plugins) to limit the intensity of FAH dynamically based on many CPU/GPU/etc. sensor parameters, if we had a FAH plugin that interacts with the FAH client. 

The installer could offer it as an optional component for those users who have stability issues or are annoyed by the noise.

The other option that comes to mind is kind of a hack, but it could be still useful: [MSI Afterburner](https://www.msi.com/Landing/afterburner/graphics-cards) lets you set a power limit on the GPU, so you can make sure its throttled enough not to get unstable.
![image](https://github.com/user-attachments/assets/a50e2a59-eba3-4fa4-9527-4d8d272550ae)

**Comment by sudo-nano** [comment]
This is a decent workaround for Windows. It would be nice if there were similar workarounds for Linux and MacOS, but they're no substitute for what should be a relatively simple feature to implement.
